californium-core/src/main/java/org/eclipse/californium/core/network/stack/BlockwiseLayer.java

```diff
@@ -397,35 +477,41 @@ public void sendResponse(final Exchange exchange, final Response response) {
 	@Override
 	public void receiveResponse(final Exchange exchange, final Response response) {
 
-		if (exchange.getRequest().isCanceled()) {
-			// do not continue fetching blocks if canceled
-			// reject (in particular for Block+Observe)
-			if (response.getType()!=Type.ACK) {
-				LOGGER.finer("rejecting blockwise transfer for canceled Exchange");
-				EmptyMessage rst = EmptyMessage.newRST(response);
-				sendEmptyMessage(exchange, rst);
-				// Matcher sets exchange as complete when RST is sent
-			}
+		if (isTransparentBlockwiseHandlingEnabled()) {
 
-		} else if (!response.hasBlockOption()) {
+			if (exchange.getRequest().isCanceled()) {
```

Comment:
I guess we need to deal with it at the Blockwise Layer because the response we received here will be a response to a request issued by the Blockwise Layer (not the application layer). But I guess we can safely move the block down to the `else` branch.